File: Fundamentals of Discrete mathematics/Java/task_3.java
public class task_3 {
    public static void main(String[] args) {
        int count = 0;
        
        for (int num = 0; num < 1000000; num++) {
            String numStr = Integer.toString(num);
            boolean hasDuplicate = false;
            
            for (int i = 1; i < numStr.length(); i++) {
                if (numStr.charAt(i - 1) == numStr.charAt(i)) {
                    hasDuplicate = true;
                    break;
                }
            }
            
            if (!hasDuplicate) {
                count++;
            }
        }
        
        System.out.println(count);
    }
}

File: Fundamentals of Discrete mathematics/Python/task_3.py
count = 0

for num in range(1_000_000):
    num_str = str(num)
    has_duplicate = False
    for i in range(1, len(num_str)):
        if num_str[i-1] == num_str[i]:
            has_duplicate = True
            break
    if not has_duplicate:
        count += 1

print(count)